Improve static site documentation, siteroot (#2704)
scala/docs.scala-lang: _overviews/scala3-scaladoc/static-site.md

Scaladoc can transform both files and directories (to organize your documentation into a tree-like structure). By default, directories have a title based on the file name and have empty content. It is possible to provide index pages for each section by creating `index.html` or `index.md` (not both) in the dedicated directory.

Before generating your static site you need to set the `-siteroot` value in your doc `scalacOptions`. The value of this is the directory that holds your docs.  The root URL for the generated documentation will also be `<site-root>`.

> The configuration is as follows:

Author: Do you think it's okay if I detail the example like this or keep the sentence as you did?
For example if you have a directory called `docs` and you'd like that to be treated as your site root :

```
.
└── docs/
    └── _docs/
        ├── index.html
        └── getting-started.html
```

Then the configuration would be as follows:

> The configuration is as follows:

Reviewer: Actually sure, let's be super explicit and just add this in here as well. It may seem a bit duplicated, but at least then it will make it super clear that `docs` here is what `<site-root>` is up above.

Scaladoc can transform both files and directories (to organize your documentation into a tree-like structure). By default, directories have a title based on the file name and have empty content. It is possible to provide index pages for each section by creating `index.html` or `index.md` (not both) in the dedicated directory.

> Before generating your static site you also need to set the `-siteroot` to the name of your `<site-root>`,which means that when you generate documentation for your project, the root URL for the generated documentation will be `<site-root>`.
> The configuration is as follows:

Reviewer: ```suggestion
Before generating your static site you need to set the `-siteroot` value in your doc `scalacOptions`. The value of this is the directory that holds your docs.  The root URL for the generated documentation will also be `<site-root>`.
The configuration is as follows:
```

Author: Do you think it's okay if I detail it like this? Or should I just keep the sentence as you did.
For example if you have a directory called `docs` and you'd like that to be treated as your site root :

```
.
└── docs/
    └── _docs/
        ├── index.html
        └── getting-started.html
```

Then the configuration would be as follows:

```
Compile / doc / scalacOptions ++= Seq("-siteroot", "docs")
```

> Note that the `Compile / doc` part of the line refers to the `doc` task in the `Compile` configuration, which generates documentation for the project's main sources. So, the `scalacOptions` set in this line will only be used during the generation of documentation for the main sources, not during compilation or testing. Refer to the settings page for more information

Reviewer: ```suggestion
Note that the `Compile / doc` part of the line refers to the `doc` task in the `Compile` configuration, which generates documentation for the project's main sources. So, the `scalacOptions` set in this line will only be used during the generation of documentation for the main sources, not during compilation or testing. Refer to the settings page for more information
```

I actually don't think we really need to specify this here. It's pretty sbt specific, so I'm not sure if we should be explaining it here or not. However, if we do keep it there is a slight mistake in here that might cause even more confusion.

> not during compilation

Actually the `Compile / ` part of this is actually scoping it to that configuration like you said earlier. So it's actually applied to the `doc` task  _during_ compilation. But again, I don't think we really need to explain those details here.

Scaladoc can transform both files and directories (to organize your documentation into a tree-like structure). By default, directories have a title based on the file name and have empty content. It is possible to provide index pages for each section by creating `index.html` or `index.md` (not both) in the dedicated directory.

Before generating your static site you need to set the `-siteroot` value in your doc `scalacOptions`. The value of this is the directory that holds your docs.  The root URL for the generated documentation will also be `<site-root>`.

> For example if you have a directory called `docs` and you'd like that to be treated as your site root :

Reviewer: ```suggestion
For example if you have a directory called `docs` and you'd like that to be treated as your site root:
```
